src/server.rs

```diff
@@ -248,7 +248,9 @@ impl LanguageServer for LspServer {
                 execute_command_provider: None,
                 experimental: None,
                 folding_range_provider: Some(
-                    lsp::FoldingRangeProviderCapability::Simple(true),
+                    lsp::FoldingRangeProviderCapability::Simple(
+                        self.options.folding,
+                    ),
```

Comment:
Yes strange that this wasn't modular before. Was it previously always true for some reason?

Reply:
This is me asking with little context for what folding capability is...lol

Reply:
The old lsp server  had a flag for enabling or disabling folding, which is an editor feature for collapsing hierarchical blocks of code (it's a feature people either love or, very correctly, hate). I couldn't see a reason why we would want to bit flip on folding, because the server can say "I support it" and then the client can decide whether it wants to use it or not. However, I found a case where it's used while verifying the client against the lsp, so I added it back in for compatibility's sake. At some point, I'll probably remove it entirely, but I'm a little timid to do that in a place with so few tests and so many ways that it can break (that's the big epic this work is addressing).

Reply:
Got it! Thanks for breaking that down.